lib/Sema/CSSimplify.cpp

```diff
@@ -11208,8 +11208,17 @@ ConstraintSystem::simplifyApplicableFnConstraint(
   if (shouldAttemptFixes()) {
     if (auto *typeVar = type2->getAs<TypeVariableType>()) {
       auto *locator = typeVar->getImpl().getLocator();
-      if (typeVar->isPlaceholder() || hasFixFor(locator))
+      if (typeVar->isPlaceholder() || hasFixFor(locator)) {
```

Comment:
Could you please do a couple of things here:

- Remove `typeVar->isPlaceholder()` check since it's always false because type variable cannot be a placeholder anymore
- Move whole `if (shouldAttemptFixes())` to just before `TypeMatchOptions subflags = getDefaultDecompositionOptions(flags);`
-  use `desugar2` instead of `simplifyType(type2);`

Reply:
> use `desugar2` instead of `simplifyType(type2);`

We need to simplify the type because `getFixedTypeRecursive` doesn’t simplify `AnyMetatypeType`s that have a type variable as instance type, so in `desugar2` the type variable won’t have been replaced by a placeholder yet.

Reply:
Hm I keep forgetting about all the nuances of getFixedTypeRecursive vs. simplifyType. Maybe we should just call getFixedTypeRecursive on an instance type if type2 was a metatype for this new check.